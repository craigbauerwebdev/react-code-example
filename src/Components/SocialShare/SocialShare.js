import {
  LinkedinIcon,
  LinkedinShareButton,
  TwitterIcon,
  TwitterShareButton,
} from "react-share";
import React, { useEffect, useState } from "react";

import ConfigService from "services/ConfigService";
import OEPAnalytics from "Components/OEPAnalytics";
import { bpMap } from "util/bpMap";
import socialShareStyles from "./scss/social-share.module.scss";
import useToggleDisplayMQ from "hooks/useToggleDisplayMQ";

const shareUrl = window.location.href;

const SocialShare = ({ isLiveStream, title, page }) => {
  const [active, setActive] = useState(false);
  const isMobile = useToggleDisplayMQ(bpMap.tablet);
  const iconSize = isMobile ? 60 : 40;

  const showTwitterShare = () =>
    (isLiveStream && ConfigService.runValues.showLivestreamTwitter) ||
    (!isLiveStream && ConfigService.runValues.showOnDemandTwitter);

  const showLinkedInShare = () =>
    (isLiveStream && ConfigService.runValues.showLivestreamLinkedIn) ||
    (!isLiveStream && ConfigService.runValues.showOnDemandLinkedIn);

  useEffect(() => {
    const outOfBoundsClickHandler = (event) => {
      if (
        active &&
        !event.target.classList.contains(socialShareStyles.socialButton)
      ) {
        setActive(false);
      }
    };
    document.addEventListener("click", outOfBoundsClickHandler);
    return () => {
      document.removeEventListener("click", outOfBoundsClickHandler);
    };
  }, [active]);

  return (
    <>
      {(showTwitterShare() || showLinkedInShare()) && (
        <div className={socialShareStyles.container} tabIndex="0">
          <OEPAnalytics
            page={page}
            componentType="Button"
            url="Share Menu Tray"
            componentName="Share Menu Tray"
          >
            <div
              className={socialShareStyles.shareButton}
              onClick={() => setActive(!active)}
            >
              <img
                className={socialShareStyles.shareIcon}
                src="/images/icons/social-share.svg"
                alt="social share"
              />
            </div>
          </OEPAnalytics>

          <div
            className={`${socialShareStyles.dropdown} ${
              active && socialShareStyles.dropdownActive
            }`}
          >
            {showTwitterShare() && (
              <OEPAnalytics
                page={page}
                componentType="Button"
                url="Twitter Share Button"
                componentName="Twitter Share Button"
              >
                <TwitterShareButton
                  className={socialShareStyles.socialButton}
                  url={shareUrl}
                  title={title}
                >
                  <TwitterIcon
                    className={socialShareStyles.socialIcon}
                    size={iconSize}
                    bgStyle={{ fill: "#0066b2" }}
                  />
                </TwitterShareButton>
              </OEPAnalytics>
            )}
            {showLinkedInShare() && (
              <OEPAnalytics
                page={page}
                componentType="Button"
                url="Linkedin Share Button"
                componentName="Linkedin Share Button"
              >
                <LinkedinShareButton
                  className={socialShareStyles.socialButton}
                  url={shareUrl}
                  title={title}
                >
                  <LinkedinIcon
                    className={socialShareStyles.socialIcon}
                    size={iconSize}
                    bgStyle={{ fill: "#0066b2" }}
                  />
                </LinkedinShareButton>
              </OEPAnalytics>
            )}
          </div>
        </div>
      )}
    </>
  );
};

export default SocialShare;
