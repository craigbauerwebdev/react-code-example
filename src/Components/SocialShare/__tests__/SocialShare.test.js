import React from "react";
import { shallow } from "enzyme";
import SocialShare from "../SocialShare";
import toJson from "enzyme-to-json";

jest.mock("Components/OEPAnalytics", () => "OEPAnalytics");

it("renders without crashing", () => {
  const comp = shallow(
    <SocialShare isLiveStream={true} title="Unit Test" page="unit test" />
  );
  expect(toJson(comp)).toMatchSnapshot();
});
