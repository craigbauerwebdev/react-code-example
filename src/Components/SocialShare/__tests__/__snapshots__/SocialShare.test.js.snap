// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <div
    className="container"
  >
    <OEPAnalytics
      componentName="Share Menu Tray"
      componentType="Button"
      page="unit test"
      url="Share Menu Tray"
    >
      <div
        className="shareButton"
        onClick={[Function]}
      >
        <img
          alt="social share"
          className="shareIcon"
          src="/images/icons/social-share.svg"
        />
      </div>
    </OEPAnalytics>
    <div
      className="dropdown false"
    >
      <OEPAnalytics
        componentName="Twitter Share Button"
        componentType="Button"
        page="unit test"
        url="Twitter Share Button"
      >
        <ForwardRef(ShareButton-twitter)
          className="socialButton"
          title="Unit Test"
          url="http://localhost/"
        >
          <Icon
            bgStyle={
              Object {
                "fill": "#0066b2",
              }
            }
            borderRadius={0}
            className="socialIcon"
            iconFillColor="white"
            size={40}
          />
        </ForwardRef(ShareButton-twitter)>
      </OEPAnalytics>
      <OEPAnalytics
        componentName="Linkedin Share Button"
        componentType="Button"
        page="unit test"
        url="Linkedin Share Button"
      >
        <ForwardRef(ShareButton-linkedin)
          className="socialButton"
          title="Unit Test"
          url="http://localhost/"
        >
          <Icon
            bgStyle={
              Object {
                "fill": "#0066b2",
              }
            }
            borderRadius={0}
            className="socialIcon"
            iconFillColor="white"
            size={40}
          />
        </ForwardRef(ShareButton-linkedin)>
      </OEPAnalytics>
    </div>
  </div>
</Fragment>
`;
